unlock_indicator.c

```diff
@@ -91,6 +94,83 @@ static double scaling_factor(void) {
     return (dpi / 96.0);
 }
 
+/*
+ * Returns the cairo_font_face_t for sans-serif
+ * if this thing returns NULL, then we're somehow on a system without a
+ * sans-serif font, or a system without fontconfig (but with cairo somehow)
+ * Reference counts and caches the font-face, so we don't have memory leaks
+ * (and so that we don't have to go through all the parsing each time)
+ */
+
+static cairo_font_face_t *get_font_face() {
+    if (cairo_font_face_get_reference_count(sans_serif)) {
+        return cairo_font_face_reference(sans_serif);
+    }
+
+    FcResult result;
+    FcBool res = FcInit();
+    if (!res) {
+        DEBUG("Fontconfig init failed. No fonts will be available...\n");
+        return NULL;
+    }
+
+    FcPattern *pattern = FcNameParse(font_face_name);
```

Comment:
Can you add some comments to the various Fc* function calls here outlining what they do? Also, is this following some sort of example or document? If so, can you add a comment with a link to it please?

Reply:
> Also, is this following some sort of example or document?

Sort of. I found some partial examples for looking up fonts/drawing text with them, but none of them properly handled memory cleanup.